### PR TITLE
add recipe for tofu

### PR DESCRIPTION
**Pull request by flothesof**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by Didou09** [comment]
Hi there, I am the lead developer of tofu, and I agree with this excellent initiative

**Comment by flothesof** [comment]
@Didou09 
This PR was marked as stale. 
A single test fails for reasons I don't understand. Could you investigate the below error message?
```

    module = _load(spec)
  File "<frozen importlib._bootstrap>", line 702, in _load
  File "<frozen importlib._bootstrap>", line 671, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 783, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/conda/staged-recipes/build_artifacts/tofu_1602317835981/test_tmp/tofu/tests/tests01_geom/tests03_core.py", line 679, in <module>
    dCams[typ][c] = cls(Name='V1000', config=dconf[typ],
  File "/home/conda/staged-recipes/build_artifacts/tofu_1602317835981/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_p/lib/python3.8/site-packages/tofu/geom/_core.py", line 3717, in __init__
    super(Rays, self).__init__(**kwdargs)
  File "/home/conda/staged-recipes/build_artifacts/tofu_1602317835981/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_p/lib/python3.8/site-packages/tofu/utils.py", line 1551, in __init__
    self._init(**kwdargs)
  File "/home/conda/staged-recipes/build_artifacts/tofu_1602317835981/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_p/lib/python3.8/site-packages/tofu/geom/_core.py", line 4156, in _init
    self.set_dconfig(calcdgeom=False, **kwdconfig)
  File "/home/conda/staged-recipes/build_artifacts/tofu_1602317835981/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_p/lib/python3.8/site-packages/tofu/geom/_core.py", line 4171, in set_dconfig
    config = self._checkformat_inputs_dconfig(config)
  File "/home/conda/staged-recipes/build_artifacts/tofu_1602317835981/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_p/lib/python3.8/site-packages/tofu/geom/_core.py", line 4033, in _checkformat_inputs_dconfig
    raise Exception(msg)
Exception: Config TestTor is missing a StructIn!
	- 

----------------------------------------------------------------------
Ran 76 tests in 59.516s

FAILED (errors=1)

```

I think we are very close to getting it to work... but it still needs work!

**Comment by flothesof** [comment]
I am able to replicate the failure on my build environment.

Steps:

- install tofu from tofu-1.4.6.tar.gz with `pip install tofu-1.4.6.tar.gz`
- extract tofu/tests/* to independent folder named `tofu-tests_only`
- run `nosetests tofu` in `tofu-tests_only`

**Comment by flothesof** [comment]
@conda-forge-admin, please ping team

Dear conda-forge team, would you please take a look at the recipe? It seems to run following the latest commit but I still have some concerns :

- I had to add a little patch for Linux : is there a better way to do what the patch does? 
- it seems to me the build runs the tests several times : is this normal? 

Thank you a lot for your help! 

Cheers
Florian

File: recipes/tofu/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+  host:
+    - python
+    - pip
+    - cython
+    - numpy
+    - setuptools
+    - openmp  # [unix and not osx]
+  run:
```
Comment (ocefpaf) [inline recipes/tofu/meta.yaml:29]: Instead of Unix and not osx you can use probably use just `linux`.

**Pull request merged**